source/slang/slang-ir-autodiff-fwd.cpp

```diff
@@ -16,6 +16,86 @@
 
 namespace Slang
 {
+    
+IRInst* emitMakeDifferentialPair(IRBuilder* builder, IRType* pairType, IRInst* primalVal, IRInst* diffVal)
```

Comment:
I didn't want to put this one in `IRBuilder` because it has fairly non-trivial logic.. `IRBuilder` has its own `emitMakeDifferentialPair` & `emitMakeDifferentialPtrPair` which emit the specific op. 
This helped function decides which one to emit based on the conformance of the primal type. We need the `DiffTypeConformanceContext` to be around, so we can't easily move this into `IRBuilder`

Reply:
This function isn't using DiffTypeConformanceContext, and feels like it fits well in IRBuilder.

Reply:
We should move into the direction where IRBuilder can do more folding and instruction selection instead of emitting exactly the opcode being requested.